diff -Naur kernel/include/acpica/acpi.h kernel/include/acpica/acpi.h
--- kernel/include/acpica/acpi.h	2018-12-13 17:16:47.000000000 +0100
+++ kernel/include/acpica/acpi.h	2019-01-07 11:32:29.210551100 +0100
@@ -53,15 +53,32 @@
  *
  * Note: The order of these include files is important.
  */
-#include "platform/acenv.h"     /* Environment-specific items */
-#include "actypes.h"            /* ACPICA data types and structures */
-#include "platform/acenvex.h"   /* Extra environment-specific items */
-#include "acnames.h"            /* Common ACPI names and strings */
-#include "acexcep.h"            /* ACPICA exceptions */
-#include "actbl.h"              /* ACPI table definitions */
-#include "acoutput.h"           /* Error output and Debug macros */
-#include "acrestyp.h"           /* Resource Descriptor structs */
-#include "acpiosxf.h"           /* OSL interfaces (ACPICA-to-OS) */
-#include "acpixf.h"             /* ACPI core subsystem external interfaces */
+
+#include <vy.h>
+
+warn_push
+no_warn(-Wunused-parameter);
+no_warn_clang(-Wmacro-redefined);
+no_warn_clang(-Wused-but-marked-unused);
+no_warn_clang(-Wincompatible-pointer-types-discards-qualifiers);
+no_warn_clang(-Winvalid-pp-token);
+no_warn_clang(-Wmissing-variable-declarations);
+no_warn_clang(-Wnull-pointer-arithmetic);
+no_warn(-Wsign-conversion);
+no_warn_clang(-Wshorten-64-to-32);
+no_warn(-Wconversion);
+no_warn_gcc(-Wdiscarded-qualifiers);
+no_warn(-Wformat);
+
+#include <acpica/platform/acenv.h>     /* Environment-specific items */
+#include <acpica/actypes.h>            /* ACPICA data types and structures */
+#include <acpica/platform/acenvex.h>   /* Extra environment-specific items */
+#include <acpica/acnames.h>            /* Common ACPI names and strings */
+#include <acpica/acexcep.h>            /* ACPICA exceptions */
+#include <acpica/actbl.h>              /* ACPI table definitions */
+#include <acpica/acoutput.h>           /* Error output and Debug macros */
+#include <acpica/acrestyp.h>           /* Resource Descriptor structs */
+#include <acpica/acpiosxf.h>           /* OSL interfaces (ACPICA-to-OS) */
+#include <acpica/acpixf.h>             /* ACPI core subsystem external interfaces */

 #endif /* __ACPI_H__ */
diff -Naur kernel/include/acpica/platform/acenv.h kernel/include/acpica/platform/acenv.h
--- kernel/include/acpica/platform/acenv.h	2018-12-13 17:16:47.000000000 +0100
+++ kernel/include/acpica/platform/acenv.h	2018-12-24 21:20:02.865147200 +0100
@@ -182,65 +182,7 @@

 #endif

-#if defined(_LINUX) || defined(__linux__)
-#include "aclinux.h"
-
-#elif defined(_APPLE) || defined(__APPLE__)
-#include "acmacosx.h"
-
-#elif defined(__DragonFly__)
-#include "acdragonfly.h"
-
-#elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
-#include "acfreebsd.h"
-
-#elif defined(__NetBSD__)
-#include "acnetbsd.h"
-
-#elif defined(__sun)
-#include "acsolaris.h"
-
-#elif defined(MODESTO)
-#include "acmodesto.h"
-
-#elif defined(NETWARE)
-#include "acnetware.h"
-
-#elif defined(_CYGWIN)
-#include "accygwin.h"
-
-#elif defined(WIN32)
-#include "acwin.h"
-
-#elif defined(WIN64)
-#include "acwin64.h"
-
-#elif defined(_WRS_LIB_BUILD)
-#include "acvxworks.h"
-
-#elif defined(__OS2__)
-#include "acos2.h"
-
-#elif defined(__HAIKU__)
-#include "achaiku.h"
-
-#elif defined(__QNX__)
-#include "acqnx.h"
-
-/*
- * EFI applications can be built with -nostdlib, in this case, it must be
- * included after including all other host environmental definitions, in
- * order to override the definitions.
- */
-#elif defined(_AED_EFI) || defined(_GNU_EFI) || defined(_EDK2_EFI)
-#include "acefi.h"
-
-#else
-
-/* Unknown environment */
-
-#error Unknown target environment
-#endif
+#include <acpi/acvineyard.h>

 /*! [End] no source code translation !*/
